pkg/storage/utils/decomposedfs/upload/upload.go

```diff
@@ -263,15 +263,13 @@ func (upload *Upload) FinishUpload(_ context.Context) error {
 		}
 	}
 
-	if upload.async {
-		// handle postprocessing asynchronously but inform there is something in progress
-		return upload.tp.Propagate(upload.Ctx, n, upload.sizeDiff)
```

Comment:
not needed anymore? where do we propagate now?

Reply:
We still propagate. I just switched the code part so we need havae the call to `upload.tp.Propagate` only once